funfact/lang/interpreter/_einop.py

```diff
@@ -85,15 +86,30 @@ def _einop(spec: str, lhs, rhs, reduction: str, pairwise: str):
         else:
             indices_rem.append(c)
 
-    dim_lhs = tuple(dim_lhs)
-    dim_rhs = tuple(dim_rhs)
-    con_ax = tuple(con_ax)
+    # broadcast dimensions
+    lhs = lhs[tuple(dim_lhs)]
+    rhs = rhs[tuple(dim_rhs)]
+
+    # blow up kron indices
+    shape_lhs = lhs.shape
+    shape_rhs = rhs.shape
+    tile_lhs = []
+    tile_rhs = []
+    for i, c in enumerate(indices_all):
+        if c in kron_spec:
+            tile_lhs.append(shape_rhs[i])
+            tile_rhs.append(shape_lhs[i])
+        else:
+            tile_lhs.append(1)
+            tile_rhs.append(1)
+    lhs = np.tile(lhs, tuple(tile_lhs))
+    rhs = np.tile(rhs, tuple(tile_rhs))
```

Comment:
I really like the `tile` approach, which saves the creation of the 1 vector. However, I guess this only works for the RHS? e.g. for LHS, we want to get `[a, b, c] -> [a, a, a, b, b, b, c, c, c]`, while tiling gives `[a, b, c, a, b, c, a, b, c]`.

Reply:
That's right! The current implementation is wrong. I'll think a bit if it's best to keep the tiling for the RHS and use the 1 vector for the LHS, or maybe change both to the 1 vector.

Reply:
It should be fixed now. New implementation uses broadcasting for both Kronecker and free indices and does a reshape at the end to get the Kronecker ordering.